src/main/java/com/google/sps/servlets/Query.java

```diff
@@ -59,18 +113,83 @@ public void doPost(HttpServletRequest servletRequest, HttpServletResponse servle
         String type = "restaurant";
         String searchTerms = servletRequest.getParameter("searchTerms");
         String urlStr = "https://maps.googleapis.com/maps/api/place/nearbysearch/json?location=" + lat + "," + lon + "&radius=" + radius + "&type=" + type + "&keyword=" + searchTerms + "&key=" + apiKey;
+>>>>>>> resultsPage
 
         URL url = new URL(urlStr);
         URLConnection requestURL = url.openConnection();
         requestURL.connect();
 
         JsonParser jp = new JsonParser();
         JsonElement jsonElement = jp.parse(new InputStreamReader((InputStream) requestURL.getContent()));
+<<<<<<< HEAD
+        JsonObject jsonObj = jsonElement.getAsJsonObject();
+        System.out.println("json " + jsonObj.get("results"));
+        return jsonObj.get("results");
+    }
+
+    private JsonObject chooseRestaurant(JsonArray restaurants, Double price, Double rating){
+        Map<String, Integer> restaurantScores = new HashMap<>();
+        Collection<JsonObject> restaurantCollection = new Gson().fromJson(restaurants, new TypeToken<Collection<JsonObject>>(){}.getType());
+        int score;
+        int total = 0;
+
+        //Weights for how closely the searches follow the parameters
+        for(JsonObject restaurant : restaurantCollection){
+            score = 1;
+            try{
+                if(Double.parseDouble(restaurant.get("price_level").getAsString()) == price || price == 0)
```

Comment:
what about a switch?

First check if price == 0
Then store the diff of (Double.parseDouble(restaurant.get("price_level").getAsString()) - price)
Then you can switch on the diff